core/services/vrf/listener_v1.go

```diff
@@ -366,43 +384,46 @@ func (lsn *listenerV1) ProcessRequest(req *solidity_vrf_coordinator_interface.VR
 			"publicKey":     lsn.job.VRFSpec.PublicKey[:],
 		},
 		"jobRun": map[string]interface{}{
-			"logBlockHash":   req.Raw.BlockHash[:],
-			"logBlockNumber": req.Raw.BlockNumber,
-			"logTxHash":      req.Raw.TxHash,
-			"logTopics":      req.Raw.Topics,
-			"logData":        req.Raw.Data,
+			"logBlockHash":   req.req.Raw.BlockHash[:],
+			"logBlockNumber": req.req.Raw.BlockNumber,
+			"logTxHash":      req.req.Raw.TxHash,
+			"logTopics":      req.req.Raw.Topics,
+			"logData":        req.req.Raw.Data,
 		},
 	})
 
 	run := pipeline.NewRun(*lsn.job.PipelineSpec, vars)
 	// The VRF pipeline has no async tasks, so we don't need to check for `incomplete`
 	if _, err = lsn.pipelineRunner.Run(context.Background(), &run, lsn.l, true, func(tx pg.Queryer) error {
 		// Always mark consumed regardless of whether the proof failed or not.
-		if err = lsn.logBroadcaster.MarkConsumed(lb, pg.WithQueryer(tx)); err != nil {
+		if err = lsn.logBroadcaster.MarkConsumed(req.lb, pg.WithQueryer(tx)); err != nil {
 			lsn.l.Errorw("Failed mark consumed", "err", err)
 		}
 		return nil
 	}); err != nil {
 		lsn.l.Errorw("Failed executing run",
 			"err", err,
-			"reqID", hex.EncodeToString(req.RequestID[:]),
-			"reqTxHash", req.Raw.TxHash)
+			"reqID", hex.EncodeToString(req.req.RequestID[:]),
+			"reqTxHash", req.req.Raw.TxHash)
+		return false
 	} else {
 		if run.HasErrors() || run.HasFatalErrors() {
 			lsn.l.Error("VRFV1 pipeline run failed with errors",
-				"reqID", hex.EncodeToString(req.RequestID[:]),
-				"keyHash", hex.EncodeToString(req.KeyHash[:]),
-				"reqTxHash", req.Raw.TxHash,
+				"reqID", hex.EncodeToString(req.req.RequestID[:]),
+				"keyHash", hex.EncodeToString(req.req.KeyHash[:]),
+				"reqTxHash", req.req.Raw.TxHash,
 				"runErrors", run.AllErrors.ToError(),
 				"runFatalErrors", run.FatalErrors.ToError(),
 			)
+			return false
```

Comment:
One question about this case: does the job pipeline error out if the transaction reverts on-chain? One reason for a revert could be that the request is already fulfilled, I don't think we'd want to re-queue on that right?

Although I don't think we'd retry it more than just once, since hopefully the next time we get to it, the `Callbacks` check will return an empty seed and block number, and we'll just mark it as done.

Reply:
Nah it wouldn't error out here, we wouldn't see the revert until later when bptxm is checking receipts.